### PR TITLE
Feat(react): Closet hooks

### PR DESCRIPTION
**Pull request by pedro-gbf**
## Description

This adds the new closets hooks.

### Dependencies

None

## Checklist

- [x] The commit message follows our guidelines
- [x] Tests for the respective changes have been added
- [x] The code is commented, particularly in hard-to-understand areas
- [ ] The labels and/or milestones were added

## Disclaimer

By sending us your contributions, you are agreeing that your contribution is made subject to the terms of our [Contributor Ownership Statement](https://github.com/Farfetch/.github/blob/master/COS.md)

### CODE REVIEW

**Review by ruifcnunes — APPROVED** [review]

File: packages/react/src/users/hooks/types/useUserClosetItems.types.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import type {
+  Config,
+  GetUserClosetItemsQuery,
+} from '@farfetch/blackout-client';
+
+export type UseUserClosetOptions = {
```
Comment (boliveira) [inline packages/react/src/users/hooks/types/useUserClosetItems.types.ts:6]:
```suggestion
export type UseUserClosetItemsOptions = {
```

File: packages/react/src/users/hooks/__tests__/useUserCloset.test.ts [hunk 2]
```diff
@@ -0,0 +1,497 @@
+import { cleanup, renderHook } from '@testing-library/react';
```
Comment (boliveira) [inline packages/react/src/users/hooks/__tests__/useUserCloset.test.ts:1]: This file should be called `useUserClosetItems.test.ts`

File: packages/react/src/users/hooks/types/useUserClosetItems.types.ts (hunk 1 above)
Comment (pedro-gbf) [reply packages/react/src/users/hooks/types/useUserClosetItems.types.ts]: Changed it 👍

File: packages/react/src/users/hooks/__tests__/useUserCloset.test.ts (hunk 2 above)
Comment (pedro-gbf) [reply packages/react/src/users/hooks/__tests__/useUserCloset.test.ts]: Fixed 👍

**Pull request merged**